### PR TITLE
[stable/minio] Update minio chart for refactoring gateway mode

### PR DESCRIPTION
**Pull request by galiev**
#### What this PR does / why we need it:
- Combined parameters: s3gateway, azuregateway, gcsgateway, ossgateway, nasgateway in the parameter gateway to avoid simultaneous activation errors for incompatible parameters (for example, simultaneous activation of s3gateway.enabled = true and azuregateway.enabled = true)
-  Update chart for deploy minio gateway through the statefulset with PVC, which is necessary when turning on [minio cache](https://docs.min.io/docs/minio-disk-cache-guide.html)
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by nitisht** [comment]
Thanks for the PR @galiev , I am going to test it a little over a couple of days and then we can take this in

**Comment by nitisht** [comment]
@galiev could you please resolve the conflicts

**Comment by galiev** [comment]
Of course I did, @nitisht.

File: stable/minio/templates/statefulset.yaml [hunk 1]
```diff
@@ -55,6 +55,34 @@ spec:
...
+          "-ce",
+          "cp /tmp/config.json {{ .Values.configPath }} &&
+          /usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway oss {{ .Values.gateway.oss.endpointURL }}" ]
+          {{- end }}
+          {{- if eq .Values.gateway.type "nas" }}
+          command: [ "/bin/sh",
+          "-ce",
+          "/usr/bin/docker-entrypoint.sh minio -C {{ .Values.configPath }} gateway nas {{ .Values.mountPath }}" ]
+          {{- end }}
+          {{- else }}
```
Comment (nitisht) [inline stable/minio/templates/statefulset.yaml]: We can probably use StatefulSets always for gateways, but  PVC need to be mounted on value specified in `MINIO_CACHE_DRIVES`  for caching to work properly

Comment (galiev) [reply stable/minio/templates/statefulset.yaml]: @nitisht, thank for your comment, I added this env for gateway with persistence=enabled

Comment (nitisht) [reply stable/minio/templates/statefulset.yaml]: Thanks @galiev will test in a bit

**Comment by nitisht** [comment]
@galiev I tested and the Gateway behaviour seems to be working fine but caching still is not getting deployed as expected. Do you mind keeping only gateway cleanup changes in this PR so we can merge this and then carefully proceed to fix the cache deployment flow?

**Comment by galiev** [comment]
Thanks @nitisht for your comment, if I understood correctly, the problem was that with mode = standalone and gateway.enabled = true the variable MINIO_CACHE_DRIVES was not set, I'm fix it in https://github.com/helm/charts/pull/15749/commits/79af2cbe4884e883a32747d41769608c45f8ab69

**Comment by nitisht** [comment]
@galiev some comments

> the problem was that with mode = standalone and gateway.enabled = true the variable MINIO_CACHE_DRIVES was not set,

When MINIO_CACHE_DRIVES is set, we need to parse the value from this environment variable and then set that value in PVC, so that it gets mounted on that location. Additionally, caching is also valid for non-gateway mode. 

I would recommend removing all the cache related changes for now from this PR and just keep the gateway cleanup part. Adding all the changes in one PR is leading to complexity and difficulty in testing. 

Finally, gateway should only be deployed via either Deployment or StatefulSet, it is redundant to keep gateway code in both the files, please remove the gateway code from StatefulSet.

**Comment by galiev** [comment]
Thanks @nitisht for your comment, using Statefulset for the gateway was done for the following reasons: in the deployment I need to use one PVC for all replicas, and due to the fact that I use [local storage](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner), all replicas will be hosted on the same host, or I can use emptyDir, but then I I can not specify MINIO_CACHE_MAXUSE to limit the size of the cache due to the fact that this mount point is actively used by other pods that also use EmptyDir.

**Comment by nitisht** [comment]
any progress on this front @galiev

**Comment by galiev** [comment]
@nitisht, I'm so sorry for the delay, I made changes

**Pull request closed without merging**